srt-protocol/src/packet/mod.rs

```diff
@@ -138,4 +135,53 @@ impl Debug for Packet {
     }
 }
 
-pub type ReceivePacketResult = Result<(Packet, SocketAddr), PacketParseError>;
+#[derive(Debug)]
+pub enum ReceivePacketError {
+    Parse(PacketParseError),
+    Io(io::Error),
+}
+
+impl From<io::Error> for ReceivePacketError {
+    fn from(error: io::Error) -> Self {
+        ReceivePacketError::Io(error)
+    }
+}
+
+impl From<PacketParseError> for ReceivePacketError {
+    fn from(error: PacketParseError) -> Self {
+        ReceivePacketError::Parse(error)
+    }
+}
+
+impl fmt::Display for ReceivePacketError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        <Self as fmt::Debug>::fmt(self, f)
```

Comment:
Not crazy about this--this should probably just call display on it's child?

Reply:
Yeah, probably, I just wanted any old implementation to make the compiler happy.